File: frontend/src/types/stock.ts
export interface Stock {
  ticker: string
  stock_name: string
  description?: string
  sector?: string
  industry?: string
  current_price?: number
  closing_price?: number
  day_change_amount?: number
  day_change_percent?: number
  market_open?: number
  market_high?: number
  market_low?: number
  week_52_high?: number
  week_52_low?: number
  bid_price?: number
  bid_size?: number
  ask_price?: number
  ask_size?: number
  last_sale_price?: number
  last_sale_size?: number
  volume?: number
  average_volume?: number
  exchange?: string
  margin_requirement?: string
  dividend_frequency?: string
  dividend_yield_12month?: string
  ex_dividend_date?: string
  market_cap?: string
  market_cap_numeric?: number
  shares_outstanding?: string
  pe_ratio?: number
  earnings_call_date?: string
}

export interface SearchResponse {
  success: boolean
  results: Stock[]
  result_count: number
  generated_sql?: string
  error?: string
}

export interface SearchRequest {
  query: string
  limit?: number
}

File: backend/app/models/stock.py
from sqlalchemy import Column, String, Numeric, BigInteger, Integer, Date, Text
from app.database import Base


class Stock(Base):
    __tablename__ = "stocks"

    ticker = Column(String(10), primary_key=True)
    stock_name = Column(String(255), nullable=False)
    description = Column(Text)
    sector = Column(String(100))
    industry = Column(String(100))

    # Price data
    current_price = Column(Numeric(12, 4))
    closing_price = Column(Numeric(12, 4))
    day_change_amount = Column(Numeric(12, 4))
    day_change_percent = Column(Numeric(8, 4))

    # Market details
    market_open = Column(Numeric(12, 4))
    market_high = Column(Numeric(12, 4))
    market_low = Column(Numeric(12, 4))
    week_52_high = Column(Numeric(12, 4))
    week_52_low = Column(Numeric(12, 4))
    bid_price = Column(Numeric(12, 4))
    bid_size = Column(Integer)
    ask_price = Column(Numeric(12, 4))
    ask_size = Column(Integer)
    last_sale_price = Column(Numeric(12, 4))
    last_sale_size = Column(Integer)
    volume = Column(BigInteger)
    average_volume = Column(BigInteger)
    exchange = Column(String(20))
    margin_requirement = Column(String(20))

    # Dividends
    dividend_frequency = Column(String(50))
    dividend_yield_12month = Column(String(20))
    ex_dividend_date = Column(Date)

    # Financials
    market_cap = Column(String(20))
    market_cap_numeric = Column(BigInteger)
    shares_outstanding = Column(String(20))
    pe_ratio = Column(Numeric(10, 2))
    earnings_call_date = Column(Date)
